[leewaikin19] iP

leewaikin19 opened the pull request:
# Glados
> “Remember Before When I Was Talking About Smelly Garbage Standing Around Being Useless? That Was A Metaphor. I Was Actually Talking About You.” - GlaDOS

Glados allows you to
- track your tasks
- add deadlines
- save your tasks locally!!!

To install and use Glados, follow the steps below:
1. Download the latest release from [here](https://github.com/leewaikin19/ip/releases)
2. Double click to open the application
3. Add and/or remove tasks

## Why use Glados? 
Because it is only ~~$99.99~~ **free**, *open-source* and **lightweight**. 

Features:
- [x] Add and manage tasks
- [ ] Online syncing (TBA)
- [ ] Reminders (TBA)

Here is a snippet of the `main` method:
```java
public static void main(String[] args) {
    new Glados("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LWK19 committed:
Rename Glados.java

LWK19 committed:
Add echo function

LWK19 committed:
Add store list function

LWK19 committed:
Add Task.java class

LWK19 committed:
Add Todo, Deadline and Event classes

LWK19 committed:
Add Automated UI Testing

LWK19 committed:
Add Automated UI Testing

LWK19 committed:
Add error handling

LWK19 committed:
Add delete feature

LWK19 committed:
Add save feature

LWK19 committed:
Add DateTime parsing

LWK19 committed:
Refactor classes

LWK19 committed:
Refactor packages

LWK19 committed:
Merge branch 'add-gradle-support'

LWK19 committed:
Add gradle

LWK19 committed:
Add JUnit Tests

LWK19 committed:
Build JAR file

LWK19 committed:
Add JavaDoc

LWK19 committed:
Reformat files

LWK19 committed:
Add find feature

LWK19 committed:
Merge branch-A-CodingStandard

LWK19 committed:
Merge branch 'branch-Level-9'

awc1116 commented on src/main/java/glados/ui/Parser.java:
```diff
@@ -0,0 +1,169 @@
+     * @return LocalDateTime Parsed datetime
+     */
+    private static LocalDateTime parseDateTime(String str) {
+        String[] formats = { "d/M/yyyy", "d/M/yyyy HHmm", "d/M/yyyy hh:mm a", "d/M/yyyy hh:mm:ss a", "yyyy/M/d",
```
Would it be better for you to create an enum for your dateTimeFormat?

awc1116 commented on src/main/java/glados/tasks/Task.java:
```diff
@@ -0,0 +1,36 @@
+
+/** Task that can be stored in the task list */
+
+public class Task {
```
Would it be better to make it abstract since you don't have to initiate a Task object?

awc1116 commented on src/main/java/glados/commands/Command.java:
```diff
@@ -0,0 +1,71 @@
+     * @param ui      ui of the program
+     * @param storage local storage of the program
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
Would it be cleaner to make this an abstract method and for each individual command create a separate command class so that it would be easier for you to add, edit or remove specific commands when needed.

awc1116 commented on src/main/java/glados/commands/AddTaskCommand.java:
```diff
@@ -0,0 +1,116 @@
+import glados.exceptions.CommandException;
+
+/** Command to add a task to the list */
+public class AddTaskCommand extends Command {
```
Since Parser already checked for todo/deadline/event type, would it be easier for you to create 3 separate command classes to add each type of task? 

What if I have one more type of task with more variables? Would it take a lot of effort to change your current code to add in this task type?

e1093051 commented on src/main/java/glados/tasks/Task.java:
```diff
@@ -0,0 +1,36 @@
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
I like how you name the boolean variable so that its data type is told.

e1093051 commented on src/main/java/glados/tasks/Task.java:
```diff
@@ -0,0 +1,36 @@
+    /**
+     * Returns 'X' if task is done, ' ' otherwise
+     * 
+     * @return String
```
Would it be clearer to state what the return string represents?

e1093051 commented on src/main/java/glados/tasks/TaskList.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * Removes the ith task
```
You probably miss the period.

e1093051 commented on src/main/java/glados/tasks/TaskList.java:
```diff
@@ -0,0 +1,79 @@
+    /**
+     * Gets the ith task
+     * 
+     * @param index index
```
You probably want to have the i of the second index to be capital.

e1093051 reviewed (commented):
Found some minor mistakes, but overall is pretty good!

e1093051 commented on src/main/java/glados/ui/Glados.java:
```diff
@@ -0,0 +1,46 @@
+        }
+    }
+
+    public void run() {
```
Would it be clearer to roughly explain this public method?

LWK19 committed:
Add GUI feature

LWK19 committed:
change resources

LWK19 committed:
Use assertions

Assertions were not used at all.

Using assertions will help reduce potential bugs in the application.

Let's use assertions at points in the program to ensure variables have
values that are expected.

Assertions is a fast and reliable way to check the behaviour of the
program at smaller checkpoints.

LWK19 committed:
Improve code quality

Some code did not meet quality standards.

Meeting quality standards is important for readability.

Let's refactor the code to meet quality standards.

Changes involve improving JavaDoc comments.

LWK19 committed:
Set up CI

No CI is being used.

CI will help ensure the program works when pushing changes to the
repository.

Let's set up a CI worflow with GitHub.

GitHub workflows is an easy and convenient tool to use, since it is
integrated within GitHub.

LWK19 committed:
Improve code quality

leewaikin19 committed:
Merge pull request #1 from leewaikin19/branch-A-Assertions

Use assertions

LWK19 committed:
Merge branch 'master' into branch-A-CodeQuality

leewaikin19 committed:
Merge pull request #2 from leewaikin19/branch-A-CodeQuality

Branch a code quality

LWK19 committed:
Merge branch 'master' into branch-A-CI

leewaikin19 committed:
Merge pull request #3 from leewaikin19/branch-A-CI

Set up CI

LWK19 committed:
Add extension on help

LWK19 committed:
Add a user guide and fix code format

LWK19 committed:
Add feature to user guide
